### PR TITLE
Remove tight coupling to user bundle

### PR DESCRIPTION
**Pull request by core23**
## Subject

We don't need FOS or Sonata UserBundle, we can use the symfony UserInterface for everything.

I am targeting this branch, because this is BC.

## Changelog

```markdown
### Removed
- Removed tight coupling to user bundle
- Removed `sonata-project/user-bundle` dependency

```

### CODE REVIEW

**Comment by greg0ire** [comment]
Blocked by #562

File: src/Admin/PostAdmin.php [hunk 1]
```diff
@@ -36,6 +36,8 @@
 class PostAdmin extends AbstractAdmin
 {
     /**
+     * @deprecated since sonata-project/news-bundle 3.x, to be removed in 4.0.
+     *
      * @var UserManagerInterface
```
Comment (OskarStark) [inline src/Admin/PostAdmin.php]: UserManagerInterface|null ?

I mean what if it’s ignored on initialize?

Comment (core23) [reply src/Admin/PostAdmin.php]: Fixed

**Comment by core23** [comment]
No idea why travis is failing :/

**Comment by greg0ire** [comment]
> No idea why travis is failing :/

The test that fails references the `User` class from the user bundle, I hope this helps

**Comment by franmomu** [comment]
Could it be because [SonataUserExtension adds associations in its extension](https://github.com/sonata-project/SonataUserBundle/blob/4.x/src/DependencyInjection/SonataUserExtension.php#L231)?

**Comment by franmomu** [comment]
> > No idea why travis is failing :/
> 
> The test that fails references the `User` class from the user bundle, I hope this helps

Exactly, looks like[ if the class doesn't exist, it doesn't register anything](https://github.com/sonata-project/SonataNewsBundle/blob/3.x/src/DependencyInjection/SonataNewsExtension.php#L140).

**Pull request merged**